src/Server/Actions/RestoreExpectationsAction.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Mcustiel\Phiremock\Server\Actions;
+
+use Mcustiel\Phiremock\Server\Model\ExpectationStorage;
+use Mcustiel\Phiremock\Server\Model\RequestStorage;
+use Mcustiel\Phiremock\Server\Model\ScenarioStorage;
+use Mcustiel\PowerRoute\Actions\ActionInterface;
+use Mcustiel\PowerRoute\Common\TransactionData;
+use Psr\Log\LoggerInterface;
+
+class RestoreExpectationsAction implements ActionInterface
+{
+    /**
+     * @var \Mcustiel\Phiremock\Server\Model\ExpectationStorage
+     */
+    private $expectationStorage;
+    /**
+     * @var \Mcustiel\Phiremock\Server\Model\ExpectationStorage
+     */
+    private $expectationBackup;
+    /**
+     * @var \Mcustiel\Phiremock\Server\Model\RequestStorage
+     */
+    private $requestStorage;
+    /**
+     * @var \Mcustiel\Phiremock\Server\Model\ScenarioStorage
+     */
+    private $scenarioStorage;
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(
+        ExpectationStorage $expectationStorage,
+        ExpectationStorage $expectationBackup,
+        RequestStorage $requestStorage,
+        ScenarioStorage $scenarioStorage,
+        LoggerInterface $logger
+    ) {
+        $this->expectationStorage = $expectationStorage;
+        $this->expectationBackup = $expectationBackup;
+        $this->requestStorage = $requestStorage;
+        $this->scenarioStorage = $scenarioStorage;
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @see \Mcustiel\PowerRoute\Actions\ActionInterface::execute()
+     */
+    public function execute(TransactionData $transactionData, $argument = null)
+    {
+        $this->expectationStorage->clearExpectations();
+        $this->requestStorage->clearRequests();
+        $this->scenarioStorage->clearScenarios();
```

Comment:
I think the user wants to have clean behaviour when initial expectations are restored, and it's quicker do in one request that send 2 more. Anyway, it passes our needs as we want to have test execution so quick as possible.

Reply:
As I user I would expect this method to reload the expectation from the files. But it is cleaning the scenarios, requests and expectations. Those are unexpected secondary effects. It should only reload the files.

Reply:
I'll change issue and add description there to describe our needs

Reply:
Hey. It was not about the description, but the side effects of the feature. If I am restoring the expectations from json I want to only do that. In the future some other user of phiremock could have the case of having a expectation created through the api and then he wants to reload the json files and the other expectation is deleted or the list of requests is deleted and he did not want that, he just wanted the json files to be reloaded. So it is a side effect.
For me the best option is just reload and nothing else, the problem is to identify the expectations loaded from JSON files to only overwrite them, I will have a look at it as soon as I have some free time.

Reply:
Look for https://github.com/mcustiel/phiremock-codeception-extension/pull/20 please, it allows to provide cleanup in one call and saves ~0.2-0.4s for each test in comparation with 3 calls send for cleanup (clean counters, clean expectations, load JSON defined), with 1000 tests (its real number for a project) it saves 3-6 minutes.

Reply:
Yes. I understand the advantage in performance. 
My proposal is to have a separate url for this feature. something like `__phiremock/reset`. By sending a post or get request to it we reset everything to initial state by reseting expectations, scenarios and requests counter, and reloading json files. What do you think? 
Later we can add the put request to expectations that reloads the mappings only.